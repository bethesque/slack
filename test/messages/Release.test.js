const { Release } = require('../../lib/messages/release');
const releaseFixture = require('../fixtures/webhooks/release/non_prerelease.json');
const repoFixture = require('../fixtures/repo.json');

describe('Release rendering', () => {
  test('works', () => {
    const deploymentStatusMessage = new Release({
      release: releaseFixture,
      repository: repoFixture,
    });
    expect(deploymentStatusMessage.toJSON()).toMatchSnapshot();
  });
});
