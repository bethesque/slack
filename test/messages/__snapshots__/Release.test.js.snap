// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release rendering works 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "fallback": "bkeepers/dotenv v1.0.0 released",
      "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "bkeepers/dotenv <https://github.com/octocat/Hello-World/releases/v1.0.0|v1.0.0> released from \`master\`",
    },
  ],
}
`;
